project-base/storefront/components/Basic/GoogleMap/GoogleMap.tsx

```diff
@@ -0,0 +1,113 @@
+import { GoogleMapMarker } from './GoogleMapMarker';
+import { useDomainConfig } from 'components/providers/DomainConfigProvider';
+import GoogleMapReact from 'google-map-react';
+import getConfig from 'next/config';
+import { useEffect, useMemo, useState } from 'react';
+import { PointFeature } from 'supercluster';
+import { MapMarker, MapMarkerNullable } from 'types/map';
+import useSupercluster from 'use-supercluster';
+
+type GoogleMapProps = {
+    latitude?: string | null;
+    longitude?: string | null;
+    zoom?: number | null;
+    markers?: MapMarkerNullable[];
+    activeMarkerHandler?: (id: string) => void;
+    isDetail?: boolean;
+    closeMarkers?: boolean;
+};
+
+type MarkerProperties = {
+    cluster: boolean;
+    markerIdentifier: string;
+};
+
+const getMarkerIdentifier = (marker: MapMarker) => marker.identifier ?? `${marker.latitude}-${marker.longitude}`;
+
+const markerMapper = (marker: MapMarker): PointFeature<MarkerProperties> => ({
+    type: 'Feature' as const,
+    properties: {
+        cluster: false,
+        markerIdentifier: getMarkerIdentifier(marker),
+    },
+    geometry: {
+        type: 'Point' as const,
+        coordinates: [parseFloat(marker.longitude), parseFloat(marker.latitude)],
+    },
+});
+
+export const GoogleMap: FC<GoogleMapProps> = ({
+    latitude,
+    longitude,
+    zoom: zoomProps,
```

Comment:
Maybe you could name this prop `defaultZoom`, since it is actually the default value.

Reply:
Yeah good point.